go_lib/cloud-data/reconciler.go

```diff
@@ -274,3 +282,106 @@ func (c *Reconciler) cloudDiscoveryUnstructured(o *unstructured.Unstructured, in
 
 	return u, nil
 }
+
+func (c *Reconciler) discoveryDataReconcile(ctx context.Context) {
+	discoveryData, err := c.discoverer.DiscoveryData(ctx)
+	if err != nil {
+		c.logger.Errorf("Getting discovery data error: %v\n", err)
+		c.cloudRequestErrorMetric.WithLabelValues("discovery_data").Set(1.0)
+		return
+	}
+	c.cloudRequestErrorMetric.WithLabelValues("discovery_data").Set(0.0)
+
+	for i := 1; i <= 3; i++ {
+		if i > 1 {
+			c.logger.Infoln("Waiting 3 seconds before next attempt")
+			time.Sleep(3 * time.Second)
+		}
+
+		cctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		data, errGetting := c.k8sClient.Resource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}).Namespace("kube-system").Get(cctx, "d8-cloud-provider-discovery-data", metav1.GetOptions{})
```

Comment:
We can use normal (not dynamic) client for secrets. Lets initialize regular go-client and pass to Reconciler